apiserver/Makefile

```diff
@@ -6,12 +6,10 @@ LOCAL_CHECKS     = lint-cache-dir goimports check-copyright check-boring-ssl
 # Used so semaphore commits generated files when pins are updated
 EXTRA_FILES_TO_COMMIT=*_generated.go *_generated.*.go
 
-API_SERVER_IMAGE      ?=calico/apiserver
+# Name of the images.
+# e.g., <registry>/<name>:<tag>
+API_SERVER_IMAGE      ?=apiserver
```

Comment:
I'm not sure if this is going to be a problem or not, but this means that the image will be locally built and tagged as just "apiserver" instead of "calico/apiserver" (obviously you know that lol). One small issue this causes is a readability issue for the images while testing, we can no longer easily identify calico / tigera images.

Why can't the image name always be calico/<name>, and we just default DEV_REGISTRIES to be empty (meaning nothing is pushed if somebody calls cd)? The only thing doing this takes away is the ability to push to the defaulted docker registry, which (from our discussions) seemed like something we didn't want to do anyway.

Reply:
We can't have the name always be `calico/X` because it breaks the dev experience of tagging your own images.

i.e., `make DEV_REGISTRIES=caseydavenport image` should produce `caseydavenport/apiserver` and not `caseydavenport/calico/apiserver`. 

I think it's OK to have images of the form `apiserver:latest-amd64` as intermediaries - it won't actually cause conflicts between Calico / Enterprise because we also tag the produced images with DEV_REGISTIRES - the `apiserver:latest-amd64` image will just be an intermediate throw-away image. 

We could even have the build process run `docker rmi $(API_SERVER_IMAGE)` if we're concerned about leaving that intermediate image around, although then everywhere else that uses $(API_SERVER_IMAGE) will need to also be aware of DEV_REGISTRIES.

Reply:
OK I see, from an OS perspective this is more sound.

Reply:
I feel like it's sound for both... Is there a problem with having `apiserver:latest-amd64` meaning "the latest apiserver image that was built on this machine"?